File: beecrowd/1021 - Notas e Moedas.rs
use std::io;

fn main() {

    let mut entrada: String = String::new();

    io::stdin()
        .read_line(&mut entrada)
        .expect("Falha ao ler uma linha");

    let dinheiro: f64 = entrada.trim()
                            .parse::<f64>()
                            .expect("Falha ao interpretar como ponto flutuante");

    let notas = [100.0, 50.0, 20.0, 10.0, 5.0, 2.0]; 
    let moedas = [1.0, 0.5, 0.25, 0.10, 0.05, 0.01];

    let mut dinheiro_total = (dinheiro * 100.0) as i32;

    println!("NOTAS:");

    for valor in &notas {
        let valor_multiplicado = (valor * 100.0) as i32;
        let quantidade = dinheiro_total / valor_multiplicado;
        dinheiro_total %= valor_multiplicado;
        println!("{} nota(s) de R$ {:.2}", quantidade, valor);
    }

    println!("MOEDAS:");

    for valor in &moedas {
        let valor_multiplicado = (valor * 100.0) as i32;
        let quantidade = dinheiro_total / valor_multiplicado;
        dinheiro_total %= valor_multiplicado;
        println!("{} moeda(s) de R$ {:.2}", quantidade, valor);
    }
}

File: beecrowd/1021 - Notas e Moedas.go
package main

import "fmt"

func main() {
	var entrada float64

	fmt.Scan(&entrada)

	dinheiro := int32(entrada * 100)

	notas := []int32{10000, 5000, 2000, 1000, 500, 200}
	moedas := []int32{100, 50, 25, 10, 5, 1}

	fmt.Println("NOTAS:")

	for _, nota := range notas {
		quantidade := dinheiro / nota
		dinheiro %= nota

		fmt.Printf("%d nota(s) de R$ %d.00\n", quantidade, nota/100)
	}

	fmt.Println("MOEDAS:")

	for _, moeda := range moedas {
		quantidade := dinheiro / moeda
		dinheiro %= moeda

		fmt.Printf("%d moeda(s) de R$ %.2f\n", quantidade, float64(moeda)/100.0)
	}
}
